packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { KEY_SPACE, KEY_RETURN } from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label.
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
```

Comment:
Does this need to be a node? We should make it a string unless consumers need to provide icons or something in the label.

Reply:
We had this discussion earlier. need input from UX @neilpfeiffer  @jeremyfuksa  on this. 
to keep it aligned with other form controls i have added type as node.

Reply:
This should now be changed to a 'string' type. Since as part of a recent change https://github.com/cerner/terra-core/pull/2698, all the labels have been converted to a 'string' type.